Add flag to allow use of hosted MongoDB

BoDonkey opened the pull request:
<!-- Please don't delete this template -->

<!-- PULL REQUEST TEMPLATE -->
<!-- (Update "[ ]" to "[x]" to check a box) -->

## Summary

*Summarize the changes briefly, including which issue/ticket this resolves. If it closes an existing Github issue, include "Closes #[issue number]"*
This PR is in response to a request by a dev in Discord and developers at Brotherhood Mutual asking for better documentation for connecting to a non-local MongoDB server. Additionally, there has been some internal discussion of pointing people at the free tier of MongoDB Atlas to ease development setup. The issue is that our CLI will fail to create a user in this case since it is looking for the localhost server. Once this lands, the documentation will be updated. Closes PRO-5320

## What are the specific steps to test this change?

*For example:*
> 1. Run the website and log in as an admin
> 2. Open a piece manager modal and select several pieces
> 3. Click the "Archive" button on the top left of the manager and confirm that it should proceed
> 4. Check that all pieces have been archived properly
1. Create a MongoDB Atlas account and provision a database.
2. Create a new project using the `--uri` flag and the connection string from your account.
3. Start your project with the `APOS_MONGODB_URI` environment variable set to the Atlas connect stream and verify you can login with the credentials set during project creation.
## What kind of change does this PR introduce?
*(Check at least one)*

- [ ] Bug fix
- [X] New feature
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [X] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [X] The changelog is updated
- [1/2 ] Related documentation has been updated
- [ ] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

boutell commented on CHANGELOG.md:
```diff
@@ -1,5 +1,11 @@
+
+### Adds
+
+- Adds the `--uri` flag to pass a MongoDB server connection string allowing for initial user addition during project creation when a host server is being used.
```
Probably not the only URI we'll ever care about, so I suggest `mongodb-uri`.

BoDonkey replied:
Changed.

BoDonkey commented:
I added slightly to this PR. I thought (per conversations) that if an individual wanted to use Atlas or another hosted MongoDB solution from the get go that they would want to use that same connection string during development. So, I added the creation of an `.env` file. The starter kits will be modified to load the `.env` file using the `dotenv` package in parallel.

boutell commented on lib/commands/create.js:
```diff
@@ -131,13 +131,19 @@ module.exports = function (program) {
+      if (options.mongodbUri) {
         // Ensure the URI is properly quoted to handle special characters
-        createUserCommand = `APOS_MONGODB_URI="${options.uri}" ` + createUserCommand;
+        createUserCommand = `APOS_MONGODB_URI="${options.mongodbUri}" ` + createUserCommand;
```
Use the `shell-quote` module to do this a little more safely now that we're talking about URIs from Atlas whose contents are beyond our control.

BoDonkey replied:
Done.

boutell commented on lib/commands/create.js:
```diff
@@ -131,13 +131,19 @@ module.exports = function (program) {
+        if (!fs.existsSync('.env')) {
+          fs.writeFileSync('.env', '');
+        }
+        fs.appendFileSync('.env', `APOS_MONGODB_URI="${options.mongodbUri}\n"`);
```
This may cause the credentials to get committed to a repo, possibly a public repo, unless we're taking care to gitignore `.env`, which could cause confusion too... and, this unexpected file could drive them crazy if it overrides environment variables they actually set on a server later. So basically I think we're overstepping here although I appreciate the convenience you're trying to create with this.

BoDonkey replied:
I guess I misunderstood parts of our conversation. Removed.

boutell commented on CHANGELOG.md:
```diff
@@ -4,7 +4,8 @@
 
-- Adds the `--uri` flag to pass a MongoDB server connection string allowing for initial user addition during project creation when a host server is being used.
+- Adds the `--mongodb-uri` flag to pass a MongoDB server connection string allowing for initial user addition during project creation when a host server is being used.
+- Creates an `.env` file with the MongoDb URI variable set if it is passed through the `--mongodb-uri` flag.
```
Should be removed, right?

BoDonkey replied:
Yes, sorry.
